tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -1069,5 +1115,171 @@ public void TestIEnumerable()
             }
             Assert.Equal("01234<null>6789", actualStrings.ToString());
         }
+
+        [Fact]
+        public void TestColumnClip()
+        {
+            DataFrame df = MakeDataFrameWithNumericColumns(10);
+            BaseColumn clipped = df["Int"].Clip(3, 7);
+            Assert.Equal(3, clipped[0]);
+            Assert.Equal(3, clipped[1]);
+            Assert.Equal(3, clipped[2]);
+            Assert.Equal(7, clipped[8]);
+            Assert.Equal(7, clipped[9]);
+        }
+
+        [Fact]
+        public void TestDataFrameClip()
+        {
+            DataFrame df = MakeDataFrameWithNumericColumns(10);
```

Comment:
What happens when I call `df.Clip` on a DataFrame that has a string column?

Reply:
It'll throw a NotImplemented exception. I changed it to a NotSupportedException now

Reply:
Why doesn't it just skip the string columns?

Reply:
Yours was a better idea. I'm doing that in the latest commit now